src/Http/Http.Features/src/IHttpResponseStartFeature.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    /// <summary>
+    /// Feature to start response writing.
+    /// </summary>
+    public interface IHttpResponseStartFeature
+    {
+        /// <summary>
+        /// Starts the response by calling OnStarting() and making headers unmodifiable.
+        /// </summary>
+        /// <param name="flush"><c>True</c> if headers and request line should be flushed, otherwise <c>false</c>.</param>
+        /// <param name="cancellationToken"></param>
+        Task StartAsync(bool flush = false, CancellationToken cancellationToken = default);
```

Comment:
Even if you stick with the current/name signature, this belongs on IResponseBodyPipeFeature.

Reply:
Break? IResponseBodyPipeFeature is brand new.

Reply:
Ah I read this as the `IHttpResponseFeature`, regardless, we agreed this morning it would be another feature, not tied to the pipe feature.

Reply:
The plan is to implement this for all servers, even the ones not doing pipes.

Reply:
Why would we ever bother implementing this on servers not doing pipes? We already have to deal with the server not implementing the IResponseBodyPipeFeature. What's the issue of it not implementing this either? DefaultHttpResponse is polyfilling it anyway.

Reply:
Is there another way to make the status code/ headers locked without calling Flush?

Reply:
@jkotalik No. Write/Async, Flush/Async, or SendFileAsync are the currently supported ways to lock lock.

Reply:
So it does have a use case 🙃